02_bubble/davidbroughsmyth/main_test.go

```diff
@@ -0,0 +1,51 @@
+package main
+
+import (
+	"bytes"
+	"strconv"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestMainOutput(t *testing.T) {
+	var buf bytes.Buffer
+	out = &buf
+	main()
+	expected := strconv.Quote("[1 2 3 5]\n")
+	actual := strconv.Quote(buf.String())
+	assert.Equal(t, actual, expected, "Unexpected output from main()")
+}
+
+var tests = map[string]struct {
+	input []int
+	want  []int
+}{
+	"Reverse Sort":            {input: []int{5, 4, 3, 2, 1}, want: []int{1, 2, 3, 4, 5}},
+	"Already Sorted":          {input: []int{1, 2, 3, 4, 5}, want: []int{1, 2, 3, 4, 5}},
+	"All same":                {input: []int{8, 8, 8, 8, 8}, want: []int{8, 8, 8, 8, 8}},
+	"All zeros":               {input: []int{0, 0, 0, 0, 0}, want: []int{0, 0, 0, 0, 0}},
```

Comment:
This test is not really needed since zero is not any special value in the sort functions. It ends up being the same test as "All same". Extra tests that do not test something extra just become a maintenance burden, so best to avoid.

Reply:
Yes I see that now.